src/dmd/mtype.d

```diff
@@ -683,8 +683,17 @@ extern (C++) abstract class Type : RootObject
             }
             else if (t1n.ty == t2n.ty && t1n.implicitConvTo(t2n))
                 goto Lcovariant;
-            else if (t1n.ty == Tnull && t1n.implicitConvTo(t2n) && t1n.size() == t2n.size())
-                goto Lcovariant;
+            else if (t1n.ty == Tnull)
+            {
+                /* Runtime representation of `typeof(null)` is same as `void*`.
```

Comment:
Not sure what is meant here. Runtime representation is not the same as type equivalence or covariance.

Reply:
Yes, it is, as you must be able to read and pass around the type in the same way.  In the case of AAs, it is not read or passed around in the same way as typeof(null).